services/api/src/api/authentication.py

```diff
@@ -61,34 +63,47 @@ def auth_check(
     Returns:
         None: the dataset is authorized for the request
     """
-    if external_auth_url is None:
-        return True
-    try:
-        url = external_auth_url % dataset
-    except TypeError as e:
-        raise ValueError("external_auth_url must contain %s") from e
-    try:
-        response = requests.get(url, auth=RequestAuth(request), timeout=hf_timeout_seconds)
-    except Exception as err:
-        raise AuthCheckHubRequestError(
-            (
-                "Authentication check on the Hugging Face Hub failed or timed out. Please try again later, it's a"
-                " temporary internal issue."
-            ),
-            err,
-        ) from err
-
-    if response.status_code == 200:
-        return True
-    elif response.status_code == 401:
-        raise ExternalUnauthenticatedError(
-            "The dataset does not exist, or is not accessible without authentication (private or gated). Please check"
-            " the spelling of the dataset name or retry with authentication."
-        )
-    elif response.status_code in [403, 404]:
-        raise ExternalAuthenticatedError(
-            "The dataset does not exist, or is not accessible with the current credentials (private or gated). Please"
-            " check the spelling of the dataset name or retry with other authentication credentials."
-        )
-    else:
-        raise ValueError(f"Unexpected status code {response.status_code}")
+    with StepProfiler(method="auth_check", step="all"):
+        with StepProfiler(method="auth_check", step="prepare parameters"):
```

Comment:
Not sure if we should profile each section, maybe only the ones that are critical or we know can take some time. I wonder if this could increase the whole method response time.

Reply:
Yes, I was wondering the same. It seems evident that this part will not take time, but I decided to be exhaustive and always cover the whole method if I start, just in case I missed something so that the `all` step should be the sum of the other steps.
I'll merge it as is, and we can still change it afterward if needed.